controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -157,4 +208,28 @@ protected void shutDown() throws Exception {
             LoggerHelpers.traceLeave(log, objectId, "shutDown", traceId);
         }
     }
+
+    @VisibleForTesting
+    static class PendingSweepQueue {
+        private final ConcurrentLinkedQueue<Supplier<CompletableFuture<Void>>> workQueue = new ConcurrentLinkedQueue<>();
```

Comment:
This class has double-concurrency checks which are not needed. Suggestions for improvement:
1. make `workQueue` a simple `ArrayDeque`
2. make `isSealed` a regular `boolean`.
3. in `drainAndSeal`: change return type to `Collection<...>` and do not make a `List` anymore. All of your callsites work with `Collection`, which is implemented by `ArrayDeque`.